### PR TITLE
PMM-12880 Proper skip for MySQL TLS certs.

### PR DESCRIPTION
**Pull request by JiriCtvrtka**
PMM-12880

Link to the Feature Build: https://github.com/Percona-Lab/pmm-submodules/pull/3593

### CODE REVIEW

File: managed/models/agent_model.go [hunk 1]
```diff
@@ -339,10 +339,10 @@ func (s *Agent) DSN(service *Service, dsnParams DSNParams, tdp *DelimiterPair) s
 		cfg.Params = make(map[string]string)
 		if s.TLS {
 			switch {
-			case s.TLSSkipVerify:
-				cfg.Params["tls"] = skipVerify
 			case len(s.Files()) != 0:
 				cfg.Params["tls"] = "custom"
+			case s.TLSSkipVerify:
+				cfg.Params["tls"] = skipVerify
```
Comment (JiriCtvrtka) [inline managed/models/agent_model.go:345]: It was also mandatory to have "custom" as the first case.

File: managed/models/agent_model.go [hunk 2]
```diff
@@ -368,10 +368,10 @@ func (s *Agent) DSN(service *Service, dsnParams DSNParams, tdp *DelimiterPair) s
 		cfg.Params = make(map[string]string)
 		if s.TLS {
 			switch {
-			case s.TLSSkipVerify:
-				cfg.Params["tls"] = "skip-verify"
 			case len(s.Files()) != 0:
 				cfg.Params["tls"] = "custom"
+			case s.TLSSkipVerify:
+				cfg.Params["tls"] = skipVerify
```
Comment (JiriCtvrtka) [inline managed/models/agent_model.go:374]: It was also mandatory to have "custom" as the first case.

File: agent/tlshelpers/mysql.go [hunk 3]
```diff
@@ -37,8 +37,9 @@ func RegisterMySQLCerts(files map[string]string) error {
 
 	if ok := ca.AppendCertsFromPEM([]byte(files["tlsCa"])); ok {
 		err = mysql.RegisterTLSConfig("custom", &tls.Config{ //nolint:gosec
-			RootCAs:      ca,
-			Certificates: []tls.Certificate{cert},
+			RootCAs:            ca,
+			Certificates:       []tls.Certificate{cert},
+			InsecureSkipVerify: tlsSkipVerify,
```
Comment (JiriCtvrtka) [inline agent/tlshelpers/mysql.go:42]: Originally missed there for "custom" ones at all.

File: agent/tlshelpers/mysql.go [hunk 4]
```diff
@@ -36,9 +36,10 @@ func RegisterMySQLCerts(files map[string]string) error {
 	}
 
 	if ok := ca.AppendCertsFromPEM([]byte(files["tlsCa"])); ok {
-		err = mysql.RegisterTLSConfig("custom", &tls.Config{ //nolint:gosec
-			RootCAs:      ca,
-			Certificates: []tls.Certificate{cert},
+		err = mysql.RegisterTLSConfig("custom", &tls.Config{
+			RootCAs:            ca,
+			Certificates:       []tls.Certificate{cert},
+			InsecureSkipVerify: tlsSkipVerify, // #nosec G402
```
Comment (JiriCtvrtka) [inline agent/tlshelpers/mysql.go:42]: Originally missed there for "custom" ones at all.

File: managed/models/agent_model.go (hunk 1 above)
Comment (BupycHuk) [reply managed/models/agent_model.go:345]: could you please add it as a comment there?

Comment (BupycHuk) [reply managed/models/agent_model.go:345]: and also add that TLSSkipVerify for custom is handled in pmm-agent side

**Review by BupycHuk — APPROVED** [review]

File: managed/models/agent_model.go (hunk 1 above)
Comment (JiriCtvrtka) [reply managed/models/agent_model.go:347]: @BupycHuk Sure. Added on both places (exporter, QAN agents) :
https://github.com/percona/pmm/pull/2909/files#diff-55a051531f2bd4996142e7a3723954354b3c3e7ae921ca0e5b024178d9550f3aR341-R342

https://github.com/percona/pmm/pull/2909/files#diff-55a051531f2bd4996142e7a3723954354b3c3e7ae921ca0e5b024178d9550f3aR372-R373

File: agent/serviceinfobroker/service_info_broker.go [hunk 5]
```diff
@@ -84,12 +84,12 @@ func (sib *ServiceInfoBroker) GetInfoFromService(ctx context.Context, msg *agent
 	}
 }
 
-func (sib *ServiceInfoBroker) getMySQLInfo(ctx context.Context, dsn string, files *agentpb.TextFiles, id uint32) *agentpb.ServiceInfoResponse {
+func (sib *ServiceInfoBroker) getMySQLInfo(ctx context.Context, dsn string, files *agentpb.TextFiles, tlsSkipVerify bool, id uint32) *agentpb.ServiceInfoResponse {
```
Comment (ademidoff) [inline agent/serviceinfobroker/service_info_broker.go:87]: nit: Doesn't the linter yell at the line being too long? :)

**Review by ademidoff — APPROVED** [review]

File: agent/serviceinfobroker/service_info_broker.go (hunk 5 above)
Comment (JiriCtvrtka) [reply agent/serviceinfobroker/service_info_broker.go:87]: I dont see it between checks. Now it is 164 chars, I think lll is >180.

Comment (ademidoff) [reply agent/serviceinfobroker/service_info_broker.go:87]: You are right, it's < 170 :)

https://github.com/percona/pmm/blob/main/.golangci.yml#L57

**Pull request merged**